hooks/mutate_pod.go

```diff
@@ -50,7 +50,7 @@ func (m *podMutator) Handle(ctx context.Context, req admission.Request) admissio
 	m.log.Info("mutating pod,", "name", namespacedName, "profile", m.profileName)
 
 	po := &corev1.Pod{}
-	err := m.decoder.Decode(req, po)
+	err := admission.Decoder.Decode(*m.decoder, req, po)
```

Comment:
Decoder hanged from struct to interface. https://github.com/kubernetes-sigs/controller-runtime/pull/2736
You should change the type of `m.decoder` to `Decoder` instead of `*Decoder`.

Reply:
I re-wtite code accoding to your suggestion.